tedana/workflows/tedana.py

```diff
@@ -410,47 +424,65 @@ def tedana_workflow(data, tes, mask=None, mixm=None, ctab=None, manacc=None,
     if ctab and not mixm:
         LGR.warning('Argument "ctab" requires argument "mixm".')
         ctab = None
-    elif ctab and (manacc is None):
-        LGR.warning('Argument "ctab" requires argument "manacc".')
-        ctab = None
     elif manacc is not None and not mixm:
         LGR.warning('Argument "manacc" requires argument "mixm".')
         manacc = None
 
+    if t2smap is not None and op.isfile(t2smap):
+        t2smap = op.abspath(t2smap)
+        # Allow users to re-run on same folder
+        if t2smap != op.join(out_dir, 't2sv.nii.gz'):
+            shutil.copyfile(t2smap, op.join(out_dir, 't2sv.nii.gz'))
+            shutil.copyfile(t2smap, op.join(out_dir, op.basename(t2smap)))
+    elif t2smap is not None:
+        raise IOError('Argument "t2smap" must be an existing file.')
+
     RepLGR.info("TE-dependence analysis was performed on input data.")
-    if mask is None:
+    if mask and not t2smap:
+        # TODO: add affine check
+        LGR.info('Using user-defined mask')
+        RepLGR.info("A user-defined mask was applied to the data.")
+    elif t2smap and not mask:
+        LGR.info('Using user-defined T2* map to generate mask')
+        t2s_limited = utils.load_image(t2smap)
+        t2s_full = t2s_limited.copy()
+        mask = (t2s_limited != 0).astype(int)
```

Comment:
Are we not concerned about some ugly mask being generated if there are >0 values outside of the brain? Just trying to get a better idea of the use case. 

Perhaps it should ask users to inspect the output for sanity -  I can just imagine that...say someone puts a better t2star map into their EPI space, and interpolation or somesuch creates nonzero ringing - may be invisible when they view it but could generate very ugly mask. 

Maybe not a concern, thoughts? Perhaps line 432/463 deals with this.

Reply:
I agree it's problematic, but I feel like checking your own mask is implicitly assumed. Making something and not checking it is kind of on the user IMO.